### PR TITLE
Use Math.clamp instead of min/max

### PR DESCRIPTION
**Pull request by wendigo**
Math.clamp(value, minValue, maxValue) is easier to reason about than Math.min(Math.max(value, minValue), maxValue)

No release notes needed

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/server/remotetask/HttpRemoteTask.java [hunk 1]
```diff
@@ -701,8 +702,7 @@ boolean adjustSplitBatchSize(List<SplitAssignment> splitAssignments, long reques
                 }
             }
             if (numSplits != 0) {
-                newSplitBatchSize = (int) ((numSplits * (maxRequestSizeInBytes - requestSizeHeadroomInBytes)) / requestSize);
-                newSplitBatchSize = Math.max(guaranteedSplitsPerRequest, Math.min(maxUnacknowledgedSplits, newSplitBatchSize));
+                newSplitBatchSize = clamp(((numSplits * (maxRequestSizeInBytes - requestSizeHeadroomInBytes)) / requestSize), guaranteedSplitsPerRequest, maxUnacknowledgedSplits);
```
Comment (findepi) [inline core/trino-main/src/main/java/io/trino/server/remotetask/HttpRemoteTask.java:705]: redundant braces

File: core/trino-spi/src/main/java/io/trino/spi/block/BlockUtil.java [hunk 2]
```diff
@@ -97,7 +98,7 @@ static int calculateNewArraySize(int currentSize, int minimumSize)
         long newSize = (long) currentSize + (currentSize >> 1);
 
         // ensure new size is within bounds
-        newSize = Math.min(Math.max(newSize, minimumSize), MAX_ARRAY_SIZE);
+        newSize = clamp(newSize, minimumSize, MAX_ARRAY_SIZE);
```
Comment (findepi) [inline core/trino-spi/src/main/java/io/trino/spi/block/BlockUtil.java:101]: much easier to read!

File: plugin/trino-geospatial/src/main/java/io/trino/plugin/geospatial/BingTileFunctions.java [hunk 3]
```diff
@@ -575,8 +576,8 @@ private static void appendIntersectingSubtiles(
     private static Point tileXYToLatitudeLongitude(int tileX, int tileY, int zoomLevel)
     {
         long mapSize = mapSize(zoomLevel);
-        double x = (clip(tileX * TILE_PIXELS, 0, mapSize) / mapSize) - 0.5;
-        double y = 0.5 - (clip(tileY * TILE_PIXELS, 0, mapSize) / mapSize);
+        double x = (clamp(tileX * TILE_PIXELS, 0d, mapSize) / mapSize) - 0.5;
```
Comment (findepi) [inline plugin/trino-geospatial/src/main/java/io/trino/plugin/geospatial/BingTileFunctions.java:579]: we're clamping integer numbers, so `d` in `0d` is redundant.
cast clamp result explicitly to double

File: plugin/trino-geospatial/src/main/java/io/trino/plugin/geospatial/BingTileFunctions.java [hunk 4]
```diff
@@ -626,7 +627,7 @@ private static int longitudeToTileY(double latitude, long mapSize)
      */
     private static int axisToCoordinates(double axis, long mapSize)
     {
-        int tileAxis = (int) clip(axis * mapSize, 0, mapSize - 1);
+        int tileAxis = (int) clamp(axis * mapSize, 0d, mapSize - 1);
```
Comment (findepi) [inline plugin/trino-geospatial/src/main/java/io/trino/plugin/geospatial/BingTileFunctions.java:630]: redundant d

**Review by losipiuk — APPROVED** [review]

**Review by findepi — APPROVED** [review]

**Pull request merged**